packages/story-editor/src/components/library/panes/media/media3p/karma/mediaFetching.karma.js

```diff
@@ -258,75 +249,60 @@ xdescribe('Media3pPane fetching', () => {
     let mediaElements;
     await waitFor(
       () => {
-        mediaElements = section.querySelectorAll('[data-testid^=mediaElement]');
+        mediaElements = within(section).queryAllByTestId(/^mediaElement/);
+
+        // Continue waiting if number of elements does not match the expectedCount
         if (!mediaElements || mediaElements.length !== expectedCount) {
           throw new Error(
             `Not ready: ${mediaElements?.length} != ${expectedCount}`
           );
         }
-        jasmine.clock().tick(10);
       },
       { timeout: 5000 }
     );
     expect(mediaElements?.length).toBe(expectedCount);
   }
 
-  it('should render initial page with media3p tab button at top', async () => {
-    await fixture.snapshot();
-  });
-
   it('should render no results message', async () => {
-    spyOn(apiFetcher, 'listMedia').and.callFake(() => ({ media: [] }));
+    listMediaSpy.and.callFake(() => ({ media: [] }));
     await fixture.events.click(fixture.editor.library.media3pTab);
 
     await waitFor(() => {
       expect(
-        fixture.screen.getByText(new RegExp('^No media found$'))
+        fixture.screen.getByText(new RegExp('^No media found.$'))
       ).toBeTruthy();
     });
 
     await fixture.snapshot();
   });
 
-  it('should fetch media resources', async () => {
-    mockListMedia();
-    await fixture.events.click(fixture.editor.library.media3pTab);
-    await expectMediaElements(unsplashSection, MEDIA_PER_PAGE);
-  });
-
   it('should render categories and media resources', async () => {
-    mockListMedia();
-    mockListCategories();
-
     await fixture.events.click(fixture.editor.library.media3pTab);
 
-    await expectMediaElements(unsplashSection, MEDIA_PER_PAGE);
+    await expectMediaElements(
+      fixture.editor.library.media3p.unsplashSection,
+      MEDIA_PER_PAGE
+    );
 
     await fixture.snapshot();
   });
 
   it('should arrow navigate between category pills', async () => {
-    mockListMedia();
-    mockListCategories();
     await fixture.events.click(fixture.editor.library.media3pTab);
 
-    await fixture.events.focus(
-      fixture.querySelectorAll('[data-testid="pill"]')[0]
-    );
+    await fixture.events.focus(fixture.editor.library.media3p.filters[0]);
     expect(document.activeElement.textContent).toBe('Sustainability');
 
     await fixture.events.keyboard.press('ArrowRight');
     expect(document.activeElement.textContent).toBe('Wallpapers');
 
     await fixture.events.keyboard.press('tab');
```

Comment:
I know this was already here, but I have a small question. Sometimes we use `press('tab')` lowercase `t` and other times we use `press('Tab')` uppercase `T`. Do you know if there is any difference?

Reply:
Good question. I think the events are taken from [the puppeteer US keyboard layout](`https://github.com/puppeteer/puppeteer/blob/v1.14.0/lib/USKeyboardLayout.js`)